### PR TITLE
Add validation to process mediated payout tx published message

### PR DESCRIPTION
**Pull request by HenrikJannsen**
## Summary by CodeRabbit

* **Bug Fixes**
  * Strengthened transaction validation during trade deposit and payout processing
  * Enhanced signature verification for trade protocol messages
  * Improved serialized transaction handling and deserialization robustness

* **Refactor**
  * Consolidated validation utilities for transaction integrity checks
  * Simplified message processing code organization

* **Tests**
  * Expanded test coverage for transaction and signature validation scenarios
  * Added validation tests for edge cases and malformed data handling

### CODE REVIEW

File: core/src/main/java/bisq/core/trade/validation/TradeValidation.java [hunk 1]
```diff
@@ -188,21 +194,71 @@ public static String checkBitcoinAddress(String bitcoinAddress, BtcWalletService
...
     }
 
+    public static byte[] checkTransactionIsUnsigned(byte[] signedSerializedTransaction,
+                                                    BtcWalletService btcWalletService) {
+        checkNonEmptyBytes(signedSerializedTransaction, "signedSerializedTransaction");
+        checkNotNull(btcWalletService, "btcWalletService must not be null");
+        Transaction signedTransaction = toVerifiedTransaction(signedSerializedTransaction, btcWalletService);
+        checkArgument(signedTransaction.getInputs().stream().noneMatch(TradeValidation::hasSignatureData),
+                "signedSerializedTransaction must not be signed");
+        return signedSerializedTransaction;
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/main/java/bisq/core/trade/validation/TradeValidation.java:204]: Will address in next PR

**Comment by wodoro** [comment]
utACK
